forge/ee/lib/billing/Team.js

```diff
@@ -278,4 +278,15 @@ module.exports = function (app) {
         }
         return this.TeamType.getProperty('runtimes.limit', -1)
     }
+
+    app.db.models.Team.prototype._suspend = app.db.models.Team.prototype.suspend
+    /**
+     * Overloads the default suspend to include billing activity
+     * - deletes the team subscription (if one exists)
+     * - call the original suspend function to actually suspend the team's instances
+     */
+    app.db.models.Team.prototype.suspend = async function () {
+        await app.db.controllers.Subscription.deleteSubscription(this)
+        await this._suspend()
```

Comment:
This feels like a future bug waiting to happen. If (unrelatedly or due to a bug) deleteSubscription doesn't throw (i.e. fails to remove subscription but didnt throw) we would continue to shutdown all instances giving the appearance that the operation was successful.

Might be a good idea to have positive logic check that the subscription actually did get cancelled and throw if it did not.

Reply:
That isn't practical. Taking that to its logical conclusion, we would need to add checks after every single function call in the entire code base just in case the function didn't do what was expected of it.

Now, that said, I've realised there *is* a bug here in that this function call doesn't clear up the subscription in stripe - it just deletes the subscription locally. So this needs a different approach.